### PR TITLE
Limit upstream bandwidth to the Avatar Mixer

### PR DESCRIPTION
**Pull request by Atlante45**
Limit the upstream bandwidth to the Avatar Mixer for traits and identity data to 3 Mbps.

[Manuscript ticket](https://highfidelity.manuscript.com/f/cases/20053/Traits-updates-from-a-client-to-the-mixer-should-be-bounded-by-some-frequency-or-max-bandwidth)

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16471/HighFidelity-Beta-PR14721-8480624.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16471/HighFidelity-Beta-PR14721-8480624.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16471/HighFidelity-Beta-Interface-PR14721-8480624.dmg

**Comment by antoninahifi** [comment]
currently testing

**Comment by antoninahifi** [comment]
Pass

File: assignment-client/src/avatars/ScriptableAvatar.cpp [hunk 1]
```diff
@@ -91,6 +91,39 @@ void ScriptableAvatar::setSkeletonModelURL(const QUrl& skeletonModelURL) {
...
+        _nextTraitsSendWindow += timeUntilNextSend;
+
+        // Don't let the next send window lag behind if we're not sending a lot of data.
+        if (_nextTraitsSendWindow < now) {
+            _nextTraitsSendWindow = now;
+        }
+    }
+
+    bytesSent += AvatarData::sendAvatarDataPacket(sendAll);
+
```
Comment (SimonWalton-HiFi) [inline assignment-client/src/avatars/ScriptableAvatar.cpp]: Do we use the AvatarData bytes sent anywhere? I don't see that counted against the 3 Mbit/s.

Comment (Atlante45) [reply assignment-client/src/avatars/ScriptableAvatar.cpp]: I specifically exclude it from the limit for 2 reasons.
1) It is already limited to 1 packet and 50Hz, so about 600kbps
2) if somehow you're sending way too much traits info, we don't want to throttle position/joint updates.

**Pull request merged**